crates/shared/src/recent_block_cache.rs

```diff
@@ -158,28 +165,55 @@ where
         metrics_label: &'static str,
     ) -> Result<Self> {
         let block = block_stream.borrow().number;
-        Ok(Self {
+        let inner = Arc::new(Inner {
             mutexed: Mutex::new(Mutexed::new(
                 config.number_of_entries_to_auto_update,
                 block,
                 config.maximum_recent_block_age,
             )),
             number_of_blocks_to_cache: config.number_of_blocks_to_cache,
             fetcher: Arc::new(fetcher),
-            block_stream,
             maximum_retries: config.max_retries,
             delay_between_retries: config.delay_between_retries,
             metrics: Metrics::instance(observe::metrics::get_storage_registry()).unwrap(),
             metrics_label,
             requests: BoxRequestSharing::labelled("liquidity_fetching".into()),
-        })
+        });
+
+        let inner_cloned = Arc::downgrade(&inner);
+        tokio::task::spawn(
+            async move {
```

Comment:
nanonit: given the size and branch complexity of this nested task could be nice to factor it into a free method and keep `new` more contained.